craft_parts/parts.py

```diff
@@ -16,15 +16,95 @@
 
 """Definitions and helpers to handle parts."""
 
-import copy
 from pathlib import Path
 from typing import Any, Dict, List, Optional, Sequence, Set
 
+from pydantic import BaseModel, Field, ValidationError
+
 from craft_parts import errors
 from craft_parts.dirs import ProjectDirs
 from craft_parts.steps import Step
 
 
+class PartSpec(BaseModel):
+    """The part specification data."""
+
+    plugin: Optional[str]
+    source: Optional[str]
+    source_checksum: Optional[str] = ""
```

Comment:
A lot of these empty quotes should probably be Nones?

Reply:
Possibly, I'll investigate. The empty string default for source-* properties predates the adoption of the unmarshal pattern.

Reply:
Changed to use `None`.